services/utils.py

```diff
@@ -101,7 +102,9 @@ def get_mirror(status, id, row):
     else:
         host = user_media_url('addons')
 
-    return posixpath.join(host, str(id), row['filename'])
+    url = posixpath.join(host, str(id), row['filename'])
+    params = urllib.urlencode({'filehash': row['hash']})
+    return '{0}?{1}'.format(url, params)
```

Comment:
Only for cache busting, the security is done elsewhere. What would you recommend?